hip-lang-config.cmake.in

```diff
@@ -65,18 +65,81 @@ find_dependency(amd_comgr)
 
 include( "${CMAKE_CURRENT_LIST_DIR}/hip-lang-targets.cmake" )
 
+
+# Approach:Find LLVM directly using find_dependency()
+# IF LLVM_Successfully found, Check for VALID LLVM installation
+# using LLVM package version major no & amdgcn target check of clang binary
+# available in the LLVM install path found.
+# On Failure: If VALID LLVM installation is not found
+# Continued with old approach to use find_path() to search for LLVM include path
+# in predefined relative search Directories
+find_dependency(LLVM)
+if( LLVM_FOUND )
+  message(STATUS "Found LLVM PACKAGE in: ${LLVM_INSTALL_PREFIX}")
+  if( LLVM_VERSION_MAJOR )
+    # Check llvm package major version number to check for valid versions
+    if( ${LLVM_VERSION_MAJOR} GREATER_EQUAL 13 )
+      find_program(CLANG_TOOL_BINARY clang
+                   PATHS ${LLVM_TOOLS_BINARY_DIR} NO_DEFAULT_PATH)
+      if( CLANG_TOOL_BINARY )
+        # fetch clang targets
+        execute_process(COMMAND ${CLANG_TOOL_BINARY} --print-targets
+          OUTPUT_VARIABLE LLVM_CLANG_AMDGCN_DETAILS
+           OUTPUT_STRIP_TRAILING_WHITESPACE
+           RESULT_VARIABLE CLANG_AMDGCN_FETCH_RESULT)
+        # IF Successfully fetch clang targets, Check for amdgcn target
+        if( (NOT CLANG_AMDGCN_FETCH_RESULT) AND
+                              (LLVM_CLANG_AMDGCN_DETAILS MATCHES "amdgcn") )
+          message(STATUS "LLVM Package Clang Version ${LLVM_PACKAGE_VERSION}"
+                         " with amdgcn target FOUND" )
+          file(GLOB HIP_CLANG_INCLUDE_PATH "${LLVM_LIBRARY_DIR}/clang/*/include" )
```

Comment:
The `HIP_CLANG_INCLUDE_PATH` should be set relative to the `CMAKE_HIP_COMPILER`. This matches the behavior we do hip-config.cmake where we find the path relative to the `CMAKE_CXX_COMPILER` when using clang:

https://github.com/ROCm-Developer-Tools/hipamd/blob/develop/hip-config.cmake.in#L172

Since this is for the HIP language instead of the CXX language you should use the `CMAKE_HIP_COMPILER` instead.

Reply:
Actually @yxsamliu mentioned clang already adds these includes so all of this can be removed. We only need to keep the logic for compiler-rt.